refactor(ivy): remove instruction usage from other instructions

crisbeto opened the pull request:
Removes direct calls from one instruction into another, moves the shared logic into a separate function and removes the state getters from the shared function.

This PR resolves FW-1340.

kara commented on packages/core/src/render3/instructions/styling.ts:
```diff
@@ -146,9 +146,15 @@ function initStyling(
   const valueToAdd = resolveStylePropValue(value, suffix);
-  const stylingContext = getStylingContext(index, getLView());
+  const stylingContext = getStylingContext(selectedIndex, lView);
   const directiveStylingIndex = getActiveDirectiveStylingIndex();
```
Can we move this call up to the instruction too? It just combines two global vars. Same in `classMapInternal`

kara commented on packages/core/src/render3/instructions/listener.ts:
```diff
@@ -59,7 +59,7 @@ export function ɵɵlistener(
  * @codeGenApi
 */
-export function ɵɵcomponentHostSyntheticListener<T>(
+export function ɵɵcomponentHostSyntheticListener(
```
This change seems somewhat unrelated to the others. Why is it necessary?

crisbeto replied:
I noticed that the `T` wasn't being used while I was scanning through the file so I removed it.

kara reviewed (approved):
Thanks for this cleanup! Some nits below

crisbeto committed:
refactor(ivy): remove instruction usage from other instructions

Removes direct calls from one instruction into another, moves the shared logic into a separate function and removes the state getters from the shared function.

This PR resolves FW-1340.

crisbeto commented:
The feedback has been addressed @kara.

kara reviewed (approved):
LGTM

kara commented:
[VE presubmit](https://test.corp.google.com/ui#id=OCL:257420223:BASE:257420248:1562776362707:9f9f8560)

[Ivy presubmit](http://test/OCL:257420223:BASE:257420742:1562776536329:79c60cd0)

kara commented:
merge-assistance: codefresh is red on master + global approval
